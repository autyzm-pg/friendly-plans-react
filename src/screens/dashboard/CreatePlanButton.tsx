import React from 'react';

import { FlatButton } from 'components';
import { i18n } from 'locale';
import { Plan, Student } from 'models';
import { palette } from 'styles';

interface Props {
  student: Student;
}

export class CreatePlanButton extends React.PureComponent<Props> {
  createPlanForStudent = () => Plan.create(this.props.student);

  render() {
    return (
      <FlatButton
        title={i18n.t('planList:createPlan')}
        icon={{
          name: 'database-plus',
          type: 'material-community',
          color: palette.primaryVariant,
        }}
        onPress={this.createPlanForStudent}
      />
    );
  }
}
