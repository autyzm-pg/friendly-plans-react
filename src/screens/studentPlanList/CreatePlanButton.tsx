import { Button } from 'components';
import { i18n } from 'locale';
import React from 'react';
import { StyleSheet } from 'react-native';
import { dimensions, palette } from 'styles';

interface Props {
  onPress: () => void;
}

export class CreatePlanButton extends React.PureComponent<Props> {
  render() {
    return (
      <Button
        title={i18n.t('planList:createPlan')}
        icon={{
          name: 'addfile',
          type: 'antdesign',
          color: palette.textWhite,
          size: 13,
        }}
        buttonStyle={styles.button}
        onPress={this.props.onPress}
        titleStyle={styles.title}
      />
    );
  }
}

const styles = StyleSheet.create({
  button: {
    paddingVertical: dimensions.spacingSmall,
    paddingRight: 15,
    margin: 22,
  },
  title: {
    marginLeft: 3,
    textTransform: 'uppercase',
  },
});
