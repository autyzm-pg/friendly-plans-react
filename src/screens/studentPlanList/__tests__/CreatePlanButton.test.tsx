import { shallow } from 'enzyme';
import React from 'react';
import { CreatePlanButton } from '../CreatePlanButton';

describe('CreatePlanButton', () => {
  const props = {
    onPress: jest.fn(),
  };

  it('renders correctly', () => {
    const copyPlanButton = shallow(<CreatePlanButton {...props} />);

    expect(copyPlanButton).toMatchSnapshot();
  });
});
