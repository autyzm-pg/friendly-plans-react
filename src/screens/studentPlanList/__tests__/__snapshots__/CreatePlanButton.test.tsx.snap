// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePlanButton renders correctly 1`] = `
<Button
  backgroundColor="#6654EB"
  icon={
    Object {
      "color": "rgba(255, 255, 255, 1)",
      "name": "addfile",
      "size": 13,
      "type": "antdesign",
    }
  }
  isUppercase={true}
  onPress={[MockFunction]}
  title="Add new plan"
/>
`;
